### PR TITLE
Added support for maintaining .cmd file to re-create hardlinks and display unlinked file with special icon

### PR DESCRIPTION
**Pull request by jsbattig**
Added a couple of features:
1. User can now select an option to create and maintain a default .cmd file that allows for re-creation of all hardlinks of files dropped to a particular folder
2. Files that somehow were over written by some program deleting the original hard link, will be shown with a special icon selected by the user.

Also made the configuration utility compile in 64 bits. It required tinkering with jpeg unit in Delphi, used by JEDI.

### CODE REVIEW

File: source/ntfslinkext/IconOverlayHook.pas [hunk 1]
```diff
@@ -230,13 +241,57 @@ procedure TIconOverlayHook.LoadSettings;
...
+end;
+
 { TIconOverlayHookFactory }
 
 function TIconOverlayHookFactory.GetInstallationData: TExtensionRegistryData;
 begin
   Result.RootKey := HKEY_LOCAL_MACHINE;
   Result.BaseKey := 'Software\Microsoft\Windows\CurrentVersion\Explorer\' +
-                    'ShellIconOverlayIdentifiers\NTFSLink_' +
+                    'ShellIconOverlayIdentifiers\0NTFSLink_' +
```
Comment (miracle2k) [inline source/ntfslinkext/IconOverlayHook.pas:303]: Why this change?

File: source/configutil/uConfig.dfm [hunk 2]
```diff
@@ -195,13 +211,48 @@ object fConfig: TfConfig
...
     Anchors = [akLeft, akTop, akRight]
     Caption = 'Do not add "Link to" prefix when creating links'
     TabOrder = 10
   end
+  object SetupHardlinksCmdFile: TCheckBox
+    Left = 22
+    Top = 356
+    Width = 395
+    Height = 17
+    Hint =
```
Comment (miracle2k) [inline source/configutil/uConfig.dfm:240]: The Hints for the new controls haven't been updated.

**Comment by miracle2k** [comment]
It was a nice surprise to see this pull request. Thanks!

I know the problem you are addressing here, and despite the concept of a "broken hardlink" not being based in any filesystem realities, I can see this being something that would be a helpful invention - given the user an idea that the hardlink he created no longer exists.

I think were you are losing me is the cmd file. It seems like a rather specific solution where I don't see how it would be helpful to a generic user.

Now the broken hardlink icon is using the cmd files as a reference, but since NTFSLink already has this "Junction database" it is using for tracking symlinks, I think that could be used to track hardlinks  as well, and then implement the broken hardlink feature on top of it.

As it stands, I'd be quite reluctant to merge it.

File: source/ntfslinkext/IconOverlayHook.pas (hunk 1 above)
Comment (jsbattig) [inline source/ntfslinkext/IconOverlayHook.pas:303]: Michael, otherwise you hit the problem with applications that register
"many" custom overlays and the overlay you register doesn't get used.
Windows apparently process overlays on alphabetic order (even later version
of Windows, I'm running 2008R2 and I'm guessing 2012 has the same behavior)
and you can have up to 11 custom overlays (the OS reserves the other 4 for
a total of 15 overlays max).
By adding a zero on front is a dirty trick to get our overlay "on front".

Seba

On Thu, Jan 24, 2013 at 4:40 PM, Michael Elsdörfer <notifications@github.com

File: source/configutil/uConfig.dfm (hunk 2 above)
Comment (jsbattig) [inline source/configutil/uConfig.dfm:240]: I will fix this. I have a few other ideas, that might be specific to us,
but as usual you can choose to dump them! :)
Our use case involves in fact using GIT and allowing contractors to use
repos with only pieces of code of our main repo, so we "copy" files from
the main folder where they reside into the contractors folder, but it's
painful to maintain them on sync on one side, and they don't need to be
full folders but rather specific files. That's why the .cmd file, allows
you to re-create all hardlinks you created on a particular repo and the
re-creation cmd is versioned too. So, at anytime you are going to work with
that repo (the contractor one) you can double click the .cmd file and all
of your files there are magically refreshed reflecting the state of the
company's main repository.

I'm planning to enhance the broken hardlink feature too, to follow the
original hardlink. Now it does just a naive comparison with the number of
links, which doesn't work well because windows will follow a hardlink even
to the recycle bin. So the real hardlink may be gone, and it's not visually
reported. I'm going to fix that.

Regards

Seba

PS: If you don't want I send you pull requests for stuff might be
questionable, let me know. I'll keep doing them like options anyway, as you
never know when someone will find them handy.

On Thu, Jan 24, 2013 at 4:42 PM, Michael Elsdörfer <notifications@github.com
